diff --git a/bazel/cython_library.bzl b/bazel/cython_library.bzl
index 8e003c2246..96f7d1d323 100644
--- a/bazel/cython_library.bzl
+++ b/bazel/cython_library.bzl
@@ -71,6 +71,7 @@ def pyx_library(name, deps = [], py_deps = [], srcs = [], **kwargs):
         native.cc_binary(
             name = shared_object_name,
             srcs = [stem + ".cpp"],
+            copts = ["-Wno-unused-function", "-Wno-deprecated-declarations"],
             deps = deps + ["@local_config_python//:python_headers"],
             linkshared = 1,
         )
