diff --git a/CMakeLists.txt b/CMakeLists.txt
index 24680af..b4bbce2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -270,11 +270,4 @@ endif(UNIX)
 # Add the project directories
 
 add_subdirectory(src)
-add_subdirectory(examples)
-add_subdirectory(examples/getstarted)
-if(NATS_BUILD_STREAMING)
-  add_subdirectory(examples/stan)
-endif()
-add_subdirectory(test)
-add_subdirectory(test/dylib)
 #----------------------------
