diff --git a/generator/js_generator.cc b/generator/js_generator.cc
index 97b5844..1351a54 100644
--- a/generator/js_generator.cc
+++ b/generator/js_generator.cc
@@ -3650,6 +3650,9 @@ void Generator::GenerateFile(const GeneratorOptions& options,
 
     for (int i = 0; i < file->dependency_count(); i++) {
       const std::string& name = file->dependency(i)->name();
+      if (ModuleAlias(name) == "github_com_gogo_protobuf_gogoproto_gogo_pb") {
+        continue;
+      }
       printer->Print(
           "var $alias$ = require('$file$');\n"
           "goog.object.extend(proto, $alias$);\n",
