diff --git a/src/google/protobuf/compiler/command_line_interface.cc b/src/google/protobuf/compiler/command_line_interface.cc
index 5e9a2c418..f95684bd1 100644
--- a/src/google/protobuf/compiler/command_line_interface.cc
+++ b/src/google/protobuf/compiler/command_line_interface.cc
@@ -1730,8 +1730,8 @@ CommandLineInterface::InterpretArgument(const std::string& name,
       if (access(disk_path.c_str(), F_OK) < 0) {
         // Try the original path; it may have just happened to have a '=' in it.
         if (access(parts[i].c_str(), F_OK) < 0) {
-          std::cerr << disk_path << ": warning: directory does not exist."
-                    << std::endl;
+          // std::cerr << disk_path << ": warning: directory does not exist."
+          //           << std::endl;
         } else {
           virtual_path = "";
           disk_path = parts[i];
diff --git a/src/google/protobuf/descriptor.cc b/src/google/protobuf/descriptor.cc
index 7456d3c3d..ddcd08c17 100644
--- a/src/google/protobuf/descriptor.cc
+++ b/src/google/protobuf/descriptor.cc
@@ -10410,10 +10410,11 @@ void DescriptorBuilder::LogUnusedDependency(const FileDescriptorProto& proto,
       if (is_error) {
         AddError(unused->name(), proto, DescriptorPool::ErrorCollector::IMPORT,
                  make_error);
-      } else {
-        AddWarning(unused->name(), proto,
-                   DescriptorPool::ErrorCollector::IMPORT, make_error);
       }
+      // else {
+      // AddWarning(unused->name(), proto,
+      //           DescriptorPool::ErrorCollector::IMPORT, make_error);
+      // }
     }
   }
 }
