diff --git a/src/google/protobuf/compiler/command_line_interface.cc b/src/google/protobuf/compiler/command_line_interface.cc
index bfc6b0552..0adbe3551 100644
--- a/src/google/protobuf/compiler/command_line_interface.cc
+++ b/src/google/protobuf/compiler/command_line_interface.cc
@@ -1729,8 +1729,8 @@ CommandLineInterface::InterpretArgument(const std::string& name,
       if (access(disk_path.c_str(), F_OK) < 0) {
         // Try the original path; it may have just happened to have a '=' in it.
         if (access(parts[i].c_str(), F_OK) < 0) {
-          std::cerr << disk_path << ": warning: directory does not exist."
-                    << std::endl;
+          // std::cerr << disk_path << ": warning: directory does not exist."
+          //           << std::endl;
         } else {
           virtual_path = "";
           disk_path = parts[i];
diff --git a/src/google/protobuf/descriptor.cc b/src/google/protobuf/descriptor.cc
index c8ce218a9..f54791103 100644
--- a/src/google/protobuf/descriptor.cc
+++ b/src/google/protobuf/descriptor.cc
@@ -7854,10 +7854,11 @@ void DescriptorBuilder::LogUnusedDependency(const FileDescriptorProto& proto,
       if (is_error) {
         AddError((*it)->name(), proto, DescriptorPool::ErrorCollector::IMPORT,
                  error_message);
-      } else {
-        AddWarning((*it)->name(), proto, DescriptorPool::ErrorCollector::IMPORT,
-                   error_message);
       }
+      // else {
+      // AddWarning((*it)->name(), proto, DescriptorPool::ErrorCollector::IMPORT,
+      // error_message);
+      // }
     }
   }
 }
