diff --git a/go/private/actions/stdlib.bzl b/go/private/actions/stdlib.bzl
index abacbc80..09b2c9bf 100644
--- a/go/private/actions/stdlib.bzl
+++ b/go/private/actions/stdlib.bzl
@@ -100,7 +100,7 @@ def _build_env(go):
     ldflags = [
         option
         for option in extldflags_from_cc_toolchain(go)
-        if option not in ("-lstdc++", "-lc++", "-static") and option not in COVERAGE_OPTIONS_DENYLIST
+        if option not in ("-lstdc++", "-lc++", "-static", "-l:libstdc++.a", "-l:libc++.a", "-l:libc++abi.a") and option not in COVERAGE_OPTIONS_DENYLIST
     ]
     env.update({
         "CGO_ENABLED": "1",
