diff --git a/tensorflow/workspace1.bzl b/tensorflow/workspace1.bzl
index b6755a6e68b..9d6562457b5 100644
--- a/tensorflow/workspace1.bzl
+++ b/tensorflow/workspace1.bzl
@@ -4,7 +4,7 @@ load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
 load("@com_github_grpc_grpc//bazel:grpc_deps.bzl", "grpc_deps")
 load("@com_google_benchmark//:bazel/benchmark_deps.bzl", "benchmark_deps")
 load("@io_bazel_rules_closure//closure:defs.bzl", "closure_repositories")
-load("@local_xla//third_party/llvm:setup.bzl", "llvm_setup")
+# load("@local_xla//third_party/llvm:setup.bzl", "llvm_setup")
 load("@rules_pkg//:deps.bzl", "rules_pkg_dependencies")
 load("//third_party/android:android_configure.bzl", "android_configure")
 
@@ -15,7 +15,7 @@ def workspace(with_rules_cc = True):
     Args:
       with_rules_cc: Unused, to be removed soon.
     """
-    llvm_setup(name = "llvm-project")
+    # llvm_setup(name = "llvm-project")
     # native.register_toolchains("@local_config_python//:py_toolchain")
     rules_pkg_dependencies()
 
diff --git a/tensorflow/workspace2.bzl b/tensorflow/workspace2.bzl
index e91df06041d..ce281ccf21c 100644
--- a/tensorflow/workspace2.bzl
+++ b/tensorflow/workspace2.bzl
@@ -475,7 +475,7 @@ def _tf_repositories():
 
     # Load the raw llvm-project.  llvm does not have build rules set up by default,
     # but provides a script for setting up build rules via overlays.
-    llvm("llvm-raw")
+    # llvm("llvm-raw")
 
     # Intel openMP that is part of LLVM sources.
     tf_http_archive(
diff --git a/third_party/xla/workspace1.bzl b/third_party/xla/workspace1.bzl
index 05121708fc5..dfc5f2cedf3 100644
--- a/third_party/xla/workspace1.bzl
+++ b/third_party/xla/workspace1.bzl
@@ -4,12 +4,12 @@ load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
 load("@com_github_grpc_grpc//bazel:grpc_deps.bzl", "grpc_deps")
 load("@io_bazel_rules_closure//closure:defs.bzl", "closure_repositories")
 load("@rules_pkg//:deps.bzl", "rules_pkg_dependencies")
-load("//third_party/llvm:setup.bzl", "llvm_setup")
+# load("//third_party/llvm:setup.bzl", "llvm_setup")
 
 # buildifier: disable=unnamed-macro
 def workspace():
     """Loads a set of TensorFlow dependencies in a WORKSPACE file."""
-    llvm_setup(name = "llvm-project")
+    # llvm_setup(name = "llvm-project")
     native.register_toolchains("@local_config_python//:py_toolchain")
     rules_pkg_dependencies()
 
diff --git a/third_party/xla/workspace2.bzl b/third_party/xla/workspace2.bzl
index 9e4166034d0..d34f643bd3f 100644
--- a/third_party/xla/workspace2.bzl
+++ b/third_party/xla/workspace2.bzl
@@ -376,7 +376,7 @@ def _tf_repositories():
 
     # Load the raw llvm-project.  llvm does not have build rules set up by default,
     # but provides a script for setting up build rules via overlays.
-    llvm("llvm-raw")
+    # llvm("llvm-raw")
 
     # Intel openMP that is part of LLVM sources.
     tf_http_archive(
