diff --color -Naur tensorflow-2.9.1/tensorflow/lite/build_def.bzl tensorflow-2.9.1_new/tensorflow/lite/build_def.bzl
--- tensorflow-2.9.1/tensorflow/lite/build_def.bzl	2022-05-22 15:28:22.000000000 -0700
+++ tensorflow-2.9.1_new/tensorflow/lite/build_def.bzl	2022-06-03 23:26:11.023540768 -0700
@@ -16,6 +16,7 @@
     """Defines common compile time flags for TFLite libraries."""
     copts = [
         "-DFARMHASH_NO_CXX_STRING",
+        "-Wno-unused-but-set-variable",
     ] + select({
         clean_dep("//tensorflow:android_arm"): [
             "-mfpu=neon",
diff --color -Naur tensorflow-2.9.1/tensorflow/workspace1.bzl tensorflow-2.9.1_new/tensorflow/workspace1.bzl
--- tensorflow-2.9.1/tensorflow/workspace1.bzl	2022-05-22 15:28:22.000000000 -0700
+++ tensorflow-2.9.1_new/tensorflow/workspace1.bzl	2022-06-03 23:25:11.088519990 -0700
@@ -7,8 +7,9 @@
 load("@rules_cuda//cuda:dependencies.bzl", "rules_cuda_dependencies")
 
 def workspace():
-    native.register_toolchains("@local_config_python//:py_toolchain")
-    rules_cuda_dependencies()
+    # Disable py_toolchain, since I can't get it to work. If someone needs it in the future they'll have to figure out tensorflow's //third_party/py:python_configure.bzl system.
+    # Disable rules_cc patching for CUDA since we don't need CUDA and it doesn't work with Pixie's build system.
+    rules_cuda_dependencies(with_rules_cc=False)
 
     closure_repositories()
 
diff --color -Naur tensorflow-2.9.1/third_party/llvm/workspace.bzl tensorflow-2.9.1_new/third_party/llvm/workspace.bzl
--- tensorflow-2.9.1/third_party/llvm/workspace.bzl	2022-05-22 15:28:22.000000000 -0700
+++ tensorflow-2.9.1_new/third_party/llvm/workspace.bzl	2022-06-03 23:25:11.088519990 -0700
@@ -12,7 +12,6 @@
         sha256 = LLVM_SHA256,
         strip_prefix = "llvm-project-{commit}".format(commit = LLVM_COMMIT),
         urls = [
-            "https://storage.googleapis.com/mirror.tensorflow.org/github.com/llvm/llvm-project/archive/{commit}.tar.gz".format(commit = LLVM_COMMIT),
             "https://github.com/llvm/llvm-project/archive/{commit}.tar.gz".format(commit = LLVM_COMMIT),
         ],
         build_file = "//third_party/llvm:llvm.BUILD",
diff --color -Naur tensorflow-2.9.1/third_party/repo.bzl tensorflow-2.9.1_new/third_party/repo.bzl
--- tensorflow-2.9.1/third_party/repo.bzl	2022-05-22 15:28:22.000000000 -0700
+++ tensorflow-2.9.1_new/third_party/repo.bzl	2022-06-03 23:25:11.088519990 -0700
@@ -20,12 +20,8 @@
     Given a URL, it returns a list of the TF-mirror cache version of that URL
     and the original URL, suitable for use in `urls` field of `tf_http_archive`.
     """
-    if not url.startswith("https://"):
-        return [url]
-    return [
-        "https://storage.googleapis.com/mirror.tensorflow.org/%s" % url[8:],
-        url,
-    ]
+    # Don't use google's mirrors since some of them are broken.
+    return [url]
 
 def _get_env_var(ctx, name):
     if name in ctx.os.environ:
@@ -106,23 +102,10 @@
     labels (e.g. '@foo//:bar') or from a label created in their repository (e.g.
     'str(Label("//:bar"))').
     """
-    if len(urls) < 2:
-        fail("tf_http_archive(urls) must have redundant URLs.")
 
-    if not any([mirror in urls[0] for mirror in (
-        "mirror.tensorflow.org",
-        "mirror.bazel.build",
-        "storage.googleapis.com",
-    )]):
-        fail("The first entry of tf_http_archive(urls) must be a mirror " +
-             "URL, preferrably mirror.tensorflow.org. Even if you don't have " +
-             "permission to mirror the file, please put the correctly " +
-             "formatted mirror URL there anyway, because someone will come " +
-             "along shortly thereafter and mirror the file.")
+    # Don't check for mirrors, because for whatever reason some of the mirrors are broken.
 
     if native.existing_rule(name):
-        print("\n\033[1;33mWarning:\033[0m skipping import of repository '" +
-              name + "' because it already exists.\n")
         return
 
     _tf_http_archive(
