diff --git a/node_modules/vega-schema/src/mark.js b/node_modules/vega-schema/src/mark.js
index 1e3392e..9a32f53 100644
--- a/node_modules/vega-schema/src/mark.js
+++ b/node_modules/vega-schema/src/mark.js
@@ -71,6 +71,7 @@ const mark = object({
   clip: markclipRef,
   sort: compareRef,
   interactive: booleanOrSignal,
+  propEventsToOverlapped: booleanType,
   encode: def('encode'),
   transform: array(def('transformMark')),
   on: def('onMarkTrigger')
