diff --git a/node_modules/vega-view-transforms/src/Mark.js b/node_modules/vega-view-transforms/src/Mark.js
index 0aed2d2b..3b37cfb1 100644
--- a/node_modules/vega-view-transforms/src/Mark.js
+++ b/node_modules/vega-view-transforms/src/Mark.js
@@ -28,6 +28,7 @@ prototype.transform = function(_, pulse) {
     mark.source = this.source; // point to upstream collector
     mark.clip = _.clip;
     mark.interactive = _.interactive;
+    mark.propEventsToOverlapped = _.propEventsToOverlapped;
     this.value = mark;
   }
 
