diff --git a/node_modules/vega-parser/src/parsers/mark.js b/node_modules/vega-parser/src/parsers/mark.js
index 89c5ca8..2982320 100644
--- a/node_modules/vega-parser/src/parsers/mark.js
+++ b/node_modules/vega-parser/src/parsers/mark.js
@@ -44,6 +44,7 @@ export default function(spec, scope) {
   op = scope.add(Mark({
     markdef:     definition(spec),
     interactive: interactive(spec.interactive, scope),
+    propEventsToOverlapped: spec.propEventsToOverlapped,
     clip:        clip(spec.clip, scope),
     context:     {$context: true},
     groups:      scope.lookup(),
