diff --git a/lib/flb_libco/settings.h b/lib/flb_libco/settings.h
index 15d35d98..fa10ca14 100644
--- a/lib/flb_libco/settings.h
+++ b/lib/flb_libco/settings.h
@@ -11,6 +11,7 @@
 #ifdef _WIN32
 #define LIBCO_MPROTECT
 #endif
+#define LIBCO_MPROTECT
 
 /*[amd64]:
    Win64 only: provides a substantial speed-up, but will thrash XMM regs
diff --git a/src/flb_parser_logfmt.c b/src/flb_parser_logfmt.c
index 7eef934c..04f1959d 100644
--- a/src/flb_parser_logfmt.c
+++ b/src/flb_parser_logfmt.c
@@ -218,7 +218,7 @@ static int logfmt_parser(struct flb_parser *parser,
                         }
                     }
                 }
-                else {
+                else if (value_len > 0) {
                     (*map_size)++;
                 }
             }